### PR TITLE
DSDEGP-2523 Build boilerplate + main app skeleton.

### PR DESCRIPTION
**Pull request by danxmoran**
Mainly adapted from Clio

### CODE REVIEW

**Comment by danxmoran** [comment]
Thoughts of what should still be added:
* pullapprove YAML
* scaladocs for traits
* README

Others?

File: reaper-server/src/main/scala/org/broadinstitute/reaper/server/edge/WorkflowTracker.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  */
+trait WorkflowTracker {
+
+  /**
+    * Generate a stream of workflow IDs which are ready for reaping.
+    *
+    * Might be an infinite stream.
+    */
+  def pullWorkflowIds(): Source[WorkflowId, NotUsed]
+
```
Comment (charlesdyan) [inline reaper-server/src/main/scala/org/broadinstitute/reaper/server/edge/WorkflowTracker.scala:27]: Should there also be a markWorkflowBeingReaped or Reaping?

**Review by charlesdyan — APPROVED** [review]
Would it be valuable to have some IT skeleton in here as well?

**Comment by danxmoran** [comment]
@ApChagi I don't have a firm idea of what that skeleton could look like, so I'm going to leave it out for now. Clio's setup is still rough around the edges so this might be a good time to try something new.

File: reaper-server/src/main/scala/org/broadinstitute/reaper/server/edge/WorkflowTracker.scala (hunk 1 above)
Comment (danxmoran) [reply reaper-server/src/main/scala/org/broadinstitute/reaper/server/edge/WorkflowTracker.scala:27]: Maybe, depending on how its laid out... I'll add a TODO comment mentioning it

**Pull request merged**